src/test/java/uk/gov/hmcts/ccd/domain/service/processor/GlobalSearchProcessorServiceTest.java

```diff
@@ -0,0 +1,458 @@
+        caseData.put("TextField1", JacksonUtils.MAPPER.readTree("\""  + lastName  + "\""));
+
+        assertSearchCriteriaField("SearchPartyName", addressValue);
+    }
+
+
+    @Test
+    void checkSearchPartyAddressLinePopulatedInSearchCriteria() throws JsonProcessingException {
+        searchParty.setSearchPartyAddressLine1("SearchPartyAddress");
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String address = "MyAddress";
+        caseData.put("SearchPartyAddress", JacksonUtils.MAPPER.readTree("\""  + address  + "\""));
+
+        assertSearchCriteriaField(ADDRESS_LINE, address);
+    }
+
+    @Test
+    void checkSearchPartyAddressLinePopulatedInSearchCriteriaWithComplexField() throws JsonProcessingException {
+
+        searchParty.setSearchPartyAddressLine1("SearchParty.Address");
+
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String address = "MyAddress";
+        caseData.put("SearchParty", JacksonUtils.MAPPER.readTree("{\"Address\": \"" + address  + "\"}"));
+
+        assertSearchCriteriaField(ADDRESS_LINE, address);
+    }
+
+    @Test
+    void checkSearchPartyEMailAddressPopulatedInSearchCriteria() throws JsonProcessingException {
+
+        searchParty.setSearchPartyEmailAddress("SearchPartyEmailAddress");
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String email = "a@b.com";
+        caseData.put("SearchPartyEmailAddress", JacksonUtils.MAPPER.readTree("\""  + email  + "\""));
+
+        assertSearchCriteriaField(EMAIL_ADDRESS, email);
+    }
+
+    @Test
+    void checkSearchPartyEMailAddressPopulatedInSearchCriteriaWithComplexField() throws JsonProcessingException {
+
+        searchParty.setSearchPartyEmailAddress("SearchParty.EmailAddress");
+
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String email = "a@b.com";
+        caseData.put("SearchParty", JacksonUtils.MAPPER.readTree("{\"EmailAddress\": \"" + email  + "\"}"));
+
+        assertSearchCriteriaField(EMAIL_ADDRESS, email);
+    }
+
+    @Test
+    void checkSearchPartyPostCodePopulatedInSearchCriteria() throws JsonProcessingException {
+
+        searchParty.setSearchPartyPostCode("SearchPartyPostCode");
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String postCode = "AB1 2CD";
+        caseData.put("SearchPartyPostCode", JacksonUtils.MAPPER.readTree("\""  + postCode  + "\""));
+
+        assertSearchCriteriaField(POST_CODE, postCode);
+    }
+
+    @Test
+    void checkSearchPartyPostCodePopulatedInSearchCriteriaWithComplexField() throws JsonProcessingException {
+
+        searchParty.setSearchPartyPostCode("MySearchParty.SearchParty.PostCode.name");
+
+        caseTypeDefinition.setSearchParties(List.of(searchParty));
+
+        final String postCode = "AB1 2CD";
+
+        caseData.put("MySearchParty", JacksonUtils.MAPPER.readTree("{\n"
```

Comment:
Not sure if this would be helpful here but just bringing it to your attention.
https://github.com/hmcts/ccd-data-store-api/blob/7088707c787a8c25e68ab9983d88e982b6eda624/src/test/java/uk/gov/hmcts/ccd/TestFixtures.java#L110